hutch_python/user_load.py

```diff
@@ -50,3 +56,181 @@ def get_user_objs(load, *, ask_on_failure=True):
                 )
 
     return objs
+
+
+def update_objs(
+    obj_ns: utils.HelpfulNamespace,
+    entry: str,
+    attrs: dict,
+    fn: Callable[[BaseInterface, Union[dict, list[str]]], None]
+) -> None:
+    """
+    Helper function for updating object namespaces
+    obj_ns is a HelpfulNamespace, which may include other
+    HelpfulNamespace's.  As a result this must be semi-recursive.
+
+    Parameters
+    ----------
+    obj_ns : HelpfulNamespace
+        Namespace of all objects
+
+    entry : str
+        Name of device to edit
+
+    attrs : list[str]
+        List of attributes to add to whitelist
+
+    fn : Callable[[BaseInterface, dict], None]
+        A function that applies a change to a singular device
+    """
+    dev = getattr(obj_ns, entry, None)
+    if dev:  # look for explicit device
+        fn(dev, attrs)
+
+        return
+
+    # look for devices with class
+    for dev in obj_ns:
+        if isinstance(dev, utils.HelpfulNamespace):
+            update_objs(dev, entry, attrs, fn)
+        # a poor-man's type check, in an effort to avoid importing
+        # every device type again
+        if entry in str(type(dev)):
```

Comment:
This is a bit worrysome for devices whose classes share similar substrings

Reply:
I think the more precise way to do a soft no-import check is to check for equality against `type(dev).__name__` (hope I didn't mess up that syntax).

Do we care at all about inheritance in these? That would be the main reason to do real import loading.

Reply:
I agree that the general `if entry in name` needs refactoring - equality check seems appropriate.

I'm stuck between:
* Inheritance-level configuration makes it so easy to change settings for all devices consistently!
* Inheritance-level configuration is too confusing, will be troublesome for us to trace back settings with, and will be a pain for users

Reply:
My thought was that typical hutch-python sessions have lots of similar devices (e.g. IMS motors) that might want similar configurations.  I agree that inheritance-level configuration might not be worthwhile, but "edit all devices of a specific class" might still have a place.